parquet-hadoop/src/test/java/org/apache/parquet/hadoop/TestParquetFileWriter.java

```diff
@@ -239,6 +248,82 @@ public void testWriteRead() throws Exception {
     PrintFooter.main(new String[] {path.toString()});
   }
 
+  @Test
+  public void testWriteReadWithRecordReader() throws Exception {
```

Comment:
I think this unit test does not reproduce the original scenario. Please check my comment in the [jira](https://issues.apache.org/jira/browse/PARQUET-2078?focusedCommentId=17406621&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-17406621).

Reply:
My bad, it seems just fine.